src/model/encoding/convertFromRawToDraftState.js

```diff
@@ -210,10 +214,19 @@ const decodeRawBlocks = (
   rawState: RawDraftContentState,
   entityMap: *,
 ): BlockMap => {
-  const rawBlocks = rawState.blocks;
-
-  if (!Array.isArray(rawBlocks[0].children)) {
-    return decodeContentBlocks(rawBlocks, entityMap);
+  const isTreeRawBlock = Array.isArray(rawState.blocks[0].children);
+  const rawBlocks =
+    experimentalTreeDataSupport && !isTreeRawBlock
+      ? DraftTreeAdapter.fromRawStateToRawTreeState(rawState).blocks
+      : rawState.blocks;
```

Comment:
This is a nit, but I would find this more clear if we split out the behavior into two branches based on whether `experimentalTreeDataSupport` is true.

If it's true, we want to:
- convert all blocks into TreeBlocks if needed
- return `decodeContentBlockNodes(rawBlocks, entityMap)`

If it's false, we want to:
- convert all blocks into non-treeBlocks if needed
- return `decodeContentBlocks(rawBlocks, entityMap)`

This is mainly for my own ability to read things, let me know if you want me to take a pass at it after you land this.

Reply:
This makes total sense to me, will update this PR to the above recommendation

Reply:
updating this inside phabricator